app/lib/ClientBuilder.ts

```diff
@@ -0,0 +1,58 @@
+import fetch from 'node-fetch';
+import { ClientBuilder, createClient } from '@commercetools/sdk-client-v2';
+import { createApiBuilderFromCtpClient } from '@commercetools/platform-sdk';
+import createAuthMiddlewareForClientCredentialsFlow from '@commercetools/sdk-client-v2/dist/declarations/src/sdk-middleware-auth/client-credentials-flow';
+import createHttpMiddleware from '@commercetools/sdk-client-v2/dist/declarations/src/sdk-middleware-http/http';
+
+export const projectKey = process.env.NEXT_PUBLIC_CTP_PROJECT_KEY;
+export const clientId = process.env.NEXT_PUBLIC_CTP_CLIENT_ID;
+export const clientSecret = process.env.NEXT_PUBLIC_CTP_CLIENT_SECRET;
+export const authUrl = process.env.NEXT_PUBLIC_CTP_AUTH_URL;
+export const apiUrl = process.env.NEXT_PUBLIC_CTP_API_URL;
+
+if (!projectKey) {
+  throw new Error('CTP_PROJECT_KEY is not defined');
+}
+
+if (!clientId) {
+  throw new Error('CTP_CLIENT_ID is not defined');
+}
+
+if (!clientSecret) {
+  throw new Error('CTP_CLIENT_SECRET is not defined');
+}
+
+if (!authUrl) {
+  throw new Error('CTP_AUTH_URL is not defined');
+}
+
+if (!apiUrl) {
+  throw new Error('CTP_API_URL is not defined');
+}
+
```

Comment:
This is not the best place to check if you set env correctly. It is better to find some package/lib or to write you own script that will run before build.

Reply:
I've moved the environment variable checks to a separate file and used non-null assertions to ensure they're defined.